File: problems/uva/p100/p100.cpp
// input n
// print n
// if n = 1 then STOP
//     if n is odd then n <- 3n + 1
//     else n <- n/2
//     GOTO 2

#include <iostream>

using namespace std;


bool is_odd(int number) {
    return number%2 ? true : false; 
}


int cycle_size(int n) {
    int count = 1;  // because n is also included in the count
    while(n!=1) {
        if(is_odd(n)) {
            n = 3 * n + 1;
        } else {
            n = n / 2;
        }
        count++;
    }
    return count;
}


int main() {
    int i = 0, j = 0;
    cin >> i >> j;

    int max_cycle = 0;
    int current_count = 0;
    for(int c = i; c <= j; c++) {
        current_count = cycle_size(c);
        if(current_count > max_cycle) {
            max_cycle = current_count;
        }
    }

    cout << i << " " << j << " " << max_cycle << endl;
    return 0;
}

File: problems/uva/p100/p100.py
# input n
# print n
# if n = 1 then STOP
#     if n is odd then n <- 3n + 1
#     else n <- n/2
#     GOTO 2


i, j = map(int, input().split())

def find_cycle(x):
    cycle_elements = [x]
    while x != 1:
        if x%2:
            x = 3 * x + 1
        else:
            x = x // 2
        cycle_elements.append(x)
    return len(cycle_elements)

max_cycles = max([find_cycle(n) for n in range(i, j+1)])
print("{} {} {}".format(i, j, max_cycles))
#print (find_cycle(10))
# numbers = []
# n = int(input())

# while True:
